lib/unison-prelude/src/Unison/Util/Map.hs

```diff
@@ -41,6 +61,17 @@ swap :: (Ord b) => Map a b -> Map b a
 swap =
   Map.foldlWithKey' (\z a b -> Map.insert b a z) mempty
 
+-- | Like 'Map.insert', but returns the value as well.
+insertLookup :: Ord k => k -> v -> Map k v -> (Maybe v, Map k v)
```

Comment:
Does this return the old value or the new value?

Reply:
It returns the old value. Returning the new would be pretty pointless 😛